examples/src/main/scala/fluence/codec/examples/PureCodecExample.scala

```diff
@@ -0,0 +1,122 @@
+      assert(res.toString == "EitherT(Failure(java.lang.NumberFormatException: For input string: \"foo\"))")
+      assert(resMonad.toString == "Failure(java.lang.NumberFormatException: For input string: \"foo\")")
+    }
+
+
+    // If we really want to receive Left with the exception info when the string argument can't be parsed, a little
+    // more effort is needed. The problem we had before was that the supplied function `(x: String) => x.toInt`
+    // could throw parse exceptions and therefore was not really pure.
+    //
+    // However, we can catch exceptions in this function and return an Either, which will make it pure. Now, all we
+    // need to do is to lift this function into the Func context.
+    val str2intEitherCodec: Bijection[String, Int] = PureCodec.build(
+      PureCodec.liftFuncEither((x: String) => Either.catchNonFatal(x.toInt).left.map(e => CodecError(e.getMessage))),
+      PureCodec.liftFuncEither((x: Int) => Either.catchNonFatal(x.toString).left.map(e => CodecError(e.getMessage)))
+    )
+
+
+    // For lawful strings – those which can be parsed into an integer the behavior hasn't really changed.
+    // Note that we receive Right(...) wrapped in the supplied monadic type.
+    {
+      val res: EitherT[Option, CodecError, Int] = str2intEitherCodec.direct[Option]("1024")
+      val resMonad = res.value
+      assert(res.toString == "EitherT(Some(Right(1024)))")
+      assert(resMonad.toString == "Some(Right(1024))")
+    }
+
+
+    // However, for strings that can't be parsed, we will receive Left(...) – which is a desired behavior!
+    {
+      val res: EitherT[Option, CodecError, Int] = str2intEitherCodec.direct[Option]("bar")
+      val resMonad = res.value
+      assert(res.toString == "EitherT(Some(Left(fluence.codec.CodecError: For input string: \"bar\")))")
+      assert(resMonad.toString == "Some(Left(fluence.codec.CodecError: For input string: \"bar\"))")
+    }
+
+
+    // It's also totally possible to perform an inverse transformation: after all, a codec is a bijection.
+    {
+      val res: EitherT[Id, CodecError, String] = str2intCodec.inverse[Id](720)
+      val resMonad: Id[Either[CodecError, String]] = res.value
+      assert(res.toString == "EitherT(Right(720))")
+      assert(resMonad.toString == "Right(720)")
+    }
+
+
+    // It's also possible to pass the to-be-converted value first, but perform the actual transformation only
+    // later on (using different enclosing monads if desired). To achieve this, `pointAt` method which returns a
+    // lazily evaluated function can be used.
+    {
+      val point: Point[Int] = str2intCodec.direct.pointAt("333")
+      val resId: EitherT[Id, CodecError, Int] = point[Id]()
+      val resOption: EitherT[Option, CodecError, Int] = point[Option]()
+      assert(resId.toString == "EitherT(Right(333))")
+      assert(resOption.toString == "EitherT(Some(Right(333)))")
+    }
+
+
+    // TODO: describe `runF` and `toKleisli`
```

Comment:
Actually, it seems that `runF` example noted on the readme page is off. When I've tried to use `runF` accordingly to it, the compilation error has occurred. The code was:
```
val x = str2intCodec.direct.runF[Id]("foo")
println(x)
```

The error was:
```
[error] /Users/xdralex/Work/fluence/codec/examples/src/main/scala/fluence/codec/examples/PureCodecExample.scala:55:41: could not find implicit value for parameter F: cats.MonadError[cats.Id, _ >: fluence.codec.CodecError]
[error]     val x = str2intCodec.direct.runF[Id]("foo")
[error]                                         ^
[error] one error found
```